keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -148,3 +148,173 @@ def get_config(self):
             }
         )
         return config
+
+    def _augment(self, inputs):
+        image = inputs.get("images", None)
+        label = inputs.get("labels", None)
+        bounding_box = inputs.get("bounding_boxes", None)
+        transformation = self.get_random_transformation(
+            image=image, label=label, bounding_box=bounding_box
+        )  # pylint: disable=assignment-from-none
+        image = self.augment_image(image, transformation=transformation)
+        result = {"images": image}
+        if label is not None:
+            label = self.augment_label(label, transformation=transformation)
+            result["labels"] = label
+        if bounding_box is not None:
+            # changed augment_bounding_box requires image as well
+            # for unnormalizing bbox coordinates
+            bounding_box = self.augment_bounding_box(
+                image, bounding_box, transformation=transformation
+            )
+            result["bounding_boxes"] = bounding_box
+        return result
+
+    def augment_bounding_box(self, image, bounding_boxes, transformation):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        transformation: takes tuple for x,y transformation None if no transformation"""
+        x, y = transformation
+        if x is not None:
+            bounding_boxes = self.augment_horizontal(image, bounding_boxes, x)
+        if y is not None:
+            bounding_boxes = self.augment_vertical(image, bounding_boxes, y)
+        return tf.clip_by_value(
+            bounding_boxes,
+            clip_value_min=0.0,
+            clip_value_max=1.0,
+        )
+
+    def augment_horizontal(self, image, bounding_boxes, x):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        x: x transformation None if no transformation"""
+        height, width, _ = image.shape
+        # split bbox coordinates [N,4] to y1,x1,y2,x2 each of shape [N,1]
+        y1, x1, y2, x2 = tf.split(bounding_boxes, 4, axis=1)
+        # unnormalize bbox coordinates to image shape
+        # and tranform into [x1,y1,x2,y2,x3,y3,x4,y4]
+        new_bboxes = tf.stack(
```

Comment:
lets add a helper for this to extract the full 8 coordinate bounding box.  Please name it:

```
convert_to_extended_corners_format
```

Reply:
this utility fn will be added in utils/bounding_box.py?

Reply:
It can just be added to the random shear file for now; we can move it there later when we have a strong need.